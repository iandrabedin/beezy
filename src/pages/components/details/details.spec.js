import React from "react";
import { cleanup, render } from "@testing-library/react";
import Details from "./details";

afterEach(cleanup);

const props = {
  credits: {
    cast: [{ cast_id: 1, character: "character", name: "name" }],
    crew: [{ credit_id: 1, job: "job", name: "name" }]
  },
  movieDetails: {
    id: 270,
    title: "Title",
    poster_path: "poster_path",
    release_date: "2020-01-01",
    genres: [{ id: 18, name: "Drama" }]
  }
};

describe("Details", () => {
  it("should have the poster image", () => {
    const { getByAltText } = render(<Details {...props} />);
    expect(getByAltText("Title")).toBeDefined();
  });

  it("should have the title, genres, cast and crew", () => {
    const { getByText } = render(<Details {...props} />);
    expect(getByText("Title")).toBeDefined();
    expect(getByText("Drama")).toBeDefined();
    expect(getByText("job")).toBeDefined();
    expect(getByText("character")).toBeDefined();
  });

  it("should have the correct format date", () => {
    const { getByText } = render(<Details {...props} />);
    expect(getByText("January 1, 2020")).toBeDefined();
  });

  it("should render as expected", () => {
    const { asFragment } = render(<Details {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
