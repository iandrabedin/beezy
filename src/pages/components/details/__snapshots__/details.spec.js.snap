// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should render as expected 1`] = `
<DocumentFragment>
  <div
    class="details-content"
  >
    <div
      class="row"
    >
      <div
        class="poster col-sm-12 col-lg-6"
      >
        <img
          alt="Title"
          src="https://image.tmdb.org/t/p/w342poster_path"
        />
      </div>
      <div
        class="col-sm-12 col-lg-6"
      >
        <h1
          class="title"
        >
          Title
        </h1>
        <p
          class="subtitle"
        >
          January 1, 2020
        </p>
        <div
          class="row"
        >
          <div
            class="pills"
          >
            Drama
          </div>
        </div>
        <h3
          class="section-title"
        >
          Overview
        </h3>
        <p
          class="overview"
        />
        <h3
          class="section-title"
        >
          Cast and Crew
        </h3>
        <div
          class="row"
        >
          <div
            class="col-sm-12 col-lg-3"
          >
            <div
              class="profile"
            >
              <div
                class="profile-image"
              >
                <img
                  alt="name"
                  src="https://image.tmdb.org/t/p/w185undefined"
                />
              </div>
              <div
                class="profile-title"
              >
                name
              </div>
              <div
                class="profile-subtitle"
              >
                character
              </div>
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-sm-12 col-lg-3"
          >
            <div
              class="profile"
            >
              <div
                class="profile-image"
              >
                <img
                  alt="name"
                  src="https://image.tmdb.org/t/p/w185undefined"
                />
              </div>
              <div
                class="profile-title"
              >
                name
              </div>
              <div
                class="profile-subtitle"
              >
                job
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
